### PR TITLE
Replace usage of `os` in src/core.py with `pathlib`

### PR DESCRIPTION
**Pull request by Celeo**
Pretty straightforward. Could use some more testing; I'm not able to really run the import functionality through its paces.

Gave up on poetry; switched to good ol' venv.

image

Also, I don't know why `to_upper` doesn't actually have an effect, but it's that way before and after.

https://github.com/samarium-lang/Samarium/issues/30

### CODE REVIEW

File: src/samarium/core.py [hunk 1]
```diff
@@ -67,17 +67,17 @@ def import_module(data: str, reg: Registry) -> None:
...
         return
     try:
-        path = sys.argv[1][: sys.argv[1].rfind("/") + 1] or "."
+        path = Path(sys.argv[1][: sys.argv[1].rfind("/") + 1] or ".")
     except IndexError:  # REPL
-        path = os.getcwd() + "/"
+        path = Path().absolute()
 
-    if f"{name}.sm" not in os.listdir(path):
+    if f"{name}.sm" not in list(path.iterdir()):
```
Comment (trag1c) [inline src/samarium/core.py:74]: Looks like this check is incorrect (`str` not in `list[PosixPath]`?). It makes local imports not work.

File: src/samarium/core.py [hunk 2]
```diff
@@ -67,17 +67,17 @@ def import_module(data: str, reg: Registry) -> None:
...
         if name not in MODULE_NAMES:
             raise exc.SamariumImportError(f"invalid module: {name}")
-        path = os.path.join(os.path.dirname(__file__), "modules")
+        path = Path(__file__).absolute().parent / 'modules'
```
Comment (trag1c) [inline src/samarium/core.py:77]:
```suggestion
        path = Path(__file__).absolute().parent / "modules"
```

File: src/samarium/core.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def run(
     Runtime.quit_on_error = quit_on_error
     code = Transpiler(tokenize(code), reg).transpile().output
     if load_template:
-        code = readfile(f"{os.path.dirname(__file__)}/template.py").replace(
+        code = readfile(Path(__file__).resolve().parent / 'template.py').replace(
```
Comment (trag1c) [inline src/samarium/core.py:130]:
```suggestion
        code = readfile(Path(__file__).resolve().parent / "template.py").replace(
```

Also, now the typehint for `readfile` (`utils.py`) is incorrect. Btw I'm not familiar with `pathlib`, but I believe there is a way to read files without using a context manager—so maybe the entire function could be replaced with some `pathlib` operation?

**Review by trag1c — CHANGES_REQUESTED** [review]

**Comment by trag1c** [comment]
> Also, I don't know why `to_upper` doesn't actually have an effect, but it's that way before and after.

Thanks for spotting that! The issue was that slices were doing the membership checking incorrectly; already fixed.

File: src/samarium/core.py (hunk 3 above)
Comment (Celeo) [reply src/samarium/core.py]: Great point, and yes: [`Path.open`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.open).

**Review by trag1c — APPROVED** [review]

**Comment by trag1c** [comment]
Thanks for contributing! :heart:

**Pull request merged**